src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java

```diff
@@ -538,20 +560,52 @@ private void finalizeCurrentVariant(final List<VariantContext> unmergedCalls,
             return;
         }
 
-        // Find samples for dropped state and synthesize. If not arrays use GQ 60
-        final Set<String> samplesNotEncountered = Sets.difference(sampleNames, currentVariantSamplesSeen);
-        for ( final String missingSample : samplesNotEncountered ) {
-            unmergedCalls.add(createRefSiteVariantContextWithGQ(missingSample, contig, start, refAllele, inferredReferenceState.getReferenceGQ()));
-        }
-
-        final VariantContext mergedVC = variantContextMerger.merge(
-                unmergedCalls,
+        VariantContext mergedVC = variantContextMerger.merge(
+                unmergedVariantCalls,
                 new SimpleInterval(contig, (int) start, (int) start),
                 refAllele.getBases()[0],
                 true,
                 false,
                 true);
 
+
+        // Reference Sites -- first create a single VC Builder
+        final VariantContextBuilder vcWithRef = new VariantContextBuilder(mergedVC);
+
+        // create alleles (same for all genotypes)
+        List<Allele> gtAlleles = Arrays.asList(mergedVC.getReference(), mergedVC.getReference());
+
+        final GenotypesContext genotypes = GenotypesContext.copy(vcWithRef.getGenotypes());
+        GenotypeBuilder genotypeBuilder = new GenotypeBuilder();
+
+        // add known ref genotypes
+        for ( final ReferenceGenotypeInfo info : referenceCalls ) {
+            genotypeBuilder.reset(false);
+            genotypeBuilder.name(info.getSampleName());
+            genotypeBuilder.alleles(gtAlleles);
+            genotypeBuilder.GQ(info.getGQ());
+            genotypes.add(genotypeBuilder.make());
+        }
+
+        // Find samples for inferred state and synthesize
+
+        // mutates in place, so rename to be clear.
+        // we don't use this again as "samplesSeen" so we don't need a full copy
+        BitSet samplesNotEncountered = samplesSeen;
+        samplesNotEncountered.xor(sampleIdsToExtractBitSet);
+
+        // Iterate through the samples not encountered
+        for (int sampleId = samplesNotEncountered.nextSetBit(0); sampleId >= 0; sampleId = samplesNotEncountered.nextSetBit(sampleId+1)) {
```

Comment:
I find this for loop kind of confusing - might be clearer to:
        for (long sampleId : samplesNotEncountered.toLongArray()) {
(and then you wouldn't need to Long.valueOf(sampleId) on line 600.

But that might not end up scaling so well?

Reply:
yeah -- it is a little confusing, but I think what you're proposing would give back an array of longs that back the bitset, and then you're iterate through those values.  I'm going to pretend a long is 8-bits for a minute.  If you made a BitSet(8) and then set bits 0,1,2 you would get back a single long with a value of "7" (bits 11100000).

Reply:
Sorry, yeah, I *completely* misunderstood what that method does. By the name, I though it returned [0L,1L, 2L], which would be useful I think.